cmd/kubeadm/app/cmd/init.go

```diff
@@ -87,6 +87,12 @@ func NewCmdInit(out io.Writer) *cobra.Command {
 			i, err := NewInit(cfgPath, internalcfg, skipPreFlight, skipTokenPrint)
 			kubeadmutil.CheckErr(err)
 			kubeadmutil.CheckErr(i.Validate(cmd))
+
+			// TODO: remove this warning in 1.9
+			if !cmd.PersistentFlags().Lookup("token-ttl").Changed {
```

Comment:
Flags or PersistentFlags?
I think Flags aggregate everything, try looking at the source and see if you can come to the same conclusion...